### ISSUE TITLE
Spurious asterisk is inserted on new line if the previous one starts from the space(s) followed by asterisk

### ISSUE THREAD

**Issue by UR3IQO** [microsoft/vscode-cpptools#5733]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Windows 8.0
- VS Code Version: 1.46.1
- C/C++ Extension Version: 0.28.3
When editing C code expression spurious asterisk is inserted on the new line if the previous line starts from the space(s) followed by asterisk

**Steps to reproduce**
Create new .c file. Enter the following code:
```
void a(void)
{
   int x = 1 
         * 1 + 1
}
```
Put cursor before the `+` and hit enter. If there is one or more space(s) before the aterisk, spurious asterisk will be inserted on the new line. See the record:
![vscode7](https://user-images.githubusercontent.com/25171239/86745208-dd378800-c042-11ea-8aeb-ec491fd1432b.gif)

The very bad thing is that the code with the asterisk added may compile without any warnings or errors, but you will get wrong result (for example multiplication instead of addition or subtraction).

The problem roots are probably from the automatic asterisk insertion function when editing comments in the following form:
```
/*************
 *
 *
 **************/
```
If so, that function should be activated only if the comment is edited.

**Comment by sean-mcmanus** [comment]
The code that implements this is at https://github.com/microsoft/vscode-cpptools/blob/release/Extension/src/LanguageServer/languageConfig.ts#L229 if anyone wants to debug/fix it.

I debugged it, but I didn't see what was wrong with the generated regex's (and I don't have currently have time to continue investigating).

**Comment by bobbrow** [comment]
When I implemented this, I emulated what VS Code was doing for TypeScript.  I seem to recall this being a problem for TypeScript too, but since I tried it again, it looks like they have a way to disable it for the case you've shared.

I believe this is the set of rules for TypeScript: https://github.com/microsoft/vscode/blob/e5ceb315da2462b0fea70716e79cc82ce708b438/extensions/typescript-language-features/src/features/languageConfiguration.ts#L21

EDIT: I think this is the commit that changed the behavior: https://github.com/microsoft/vscode/commit/81f58337a9d6fe5a1c199417b17ba1ae150085e8#diff-5fa357ec086244b9f81089781bd6e25b.  There appears to be a new property that can be used to ensure the ` * ` is not added.

**Comment by bobbrow** [comment]
Also mentioned in #2072

**Comment by UR3IQO** [comment]
I can confirm that adding to settings `"C_Cpp.commentContinuationPatterns": []` (as recomeded in the #2072) solves the problem. Of cause it also disables this function in the comments, but that is not a big problem.

**Comment by clement-analogue** [comment]
This bug is has a bug. At new line in a comment section, the asterisk is not automatically inserted anymore.

**Comment by sean-mcmanus** [comment]
@UR3IQO Fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/1.3.0-insiders2 , except for the issue mentioned by...

@clement-analogue @elahehrashedi  I can repro the issue with

```
/**
 * 
```
2nd newline doesn't continue the "*".

**Comment by sean-mcmanus** [comment]
@clement-analogue @elahehrashedi The change at https://github.com/microsoft/vscode-cpptools/pull/7238 fixes the issue (unless we find some problem with it later).

**Comment by sean-mcmanus** [comment]
Although the original repro with one "*" is fixed, this case
```
int n = 3
    * 4
    * 5  // Enter here
```
with 2 consecutive "*" is going to cause "*" to be added to the new line -- the same bug occurs with TypeScript and even with the GitHub editor. If that is a problem you should a file a new bug against VS Code, but I'm guessing they won't fix it.

**Comment by michelleangela** [comment]
Fix is released in the 1.3.0 release version https://github.com/microsoft/vscode-cpptools/releases/tag/1.3.0
